govcd/system.go

```diff
@@ -8,11 +8,26 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"regexp"
 	"strings"
 
 	"github.com/vmware/go-vcloud-director/v2/types/v56"
 )
 
+// Simple structure to pass Edge Gateway creation parameters.
+type EdgeGatewayCreation struct {
+	ExternalNetworks          []string
+	OrgName                   string
+	VdcName                   string
+	Name                      string
+	Description               string
+	BackingConfiguration      string
+	AdvancedNetworkingEnabled bool
+}
+
+// List of allowed values for GatewayBackingConfig.
+var EGWAllowedBackingConfig = []string{"compact", "full"}
```

Comment:
API doc mentions `One of: compact, full, full4`. I see it also mentioning special endpoint https://vdc-repo.vmware.com/vmwb-repository/dcr-public/76f491b4-679c-4e1e-8428-f813d668297a/a2555a1b-22f1-4cca-b481-2a98ab874022/doc/doc/types/EdgeGatewayFormFactorType.html

Maybe it also accepts it here. Users may find it useful for more thorough size configuration.

Reply:
I tried, but the vCD doesn't agree:
```
value *errors.errorString = &errors.errorString{s:"backing configuration value 'full4' not accepted. Allowed values: [compact full] "
```

Reply:
What about removing this validation all together? It's better to pass-up the vCD error to the user. We would avoid putting ourselves into a corner if there's a supported value that our vCDs don't support, but others do.

Reply:
We do this kind of validation for API values in several places. 
I want to have this list of values to use it from Terraform provider, where we are better off shielding users from vCD error messages. So far, this works for 9.1, 9.5, and 9.7.

Reply:
This is a general comment about this:
> we are better off shielding users from vCD error messages

If the message is meaningful, we **shouldn't** be shielding the user from it. On the opposite, we should be passing the message through. The users are vCD admins they wan't visibility. I'm of course referring to the message only, but not the whole response.

Reply:
I don't understand this reasoning. I am referring at the many `Validate` clause that we have in our resources. 
For example: 

`ValidateFunc: validation.StringInSlice([]string{"AllocationVApp", "AllocationPool", "ReservationPool"}, false),`
(https://github.com/terraform-providers/terraform-provider-vcd/blob/277d63eadffac079f47cd374b50978e854ac9e0b/vcd/resource_vcd_org_vdc.go#L65)

If we want to throw bare error messages to the users, why bother checking that we are accepting the right values?
Perhaps we are talking about different things here?

Reply:
I was talking more generally.

The challenge with this particular field is that there should be more acceptable values ("full4"), though we can't prove that on our vCD instances.

Reply:
List removed

Reply:
For the sake to figure out - I would try to pass parameters using more later API version.

Reply:
There is no explicit check anymore. Wrong values will be reported back by vCD